src/main/java/com/fakng/fakngagrgtr/parser/stripe/StripeParser.java

```diff
@@ -0,0 +1,156 @@
+        String url = ((DomElement) page.getByXPath(JOBS_PAGINATION_LIST_XPATH).get(0)).getAttribute("href");
+        return Integer.parseInt(url.replaceAll("\\D+", "")) / pageSize;
+    }
+
+    private List<Vacancy> fillVacancy(List<DomNode> urls, List<DomElement> cities) {
+        List<Vacancy> vacancies = new ArrayList<>();
+
+        for (int i = 0; i < urls.size(); i++) {
+            String jobId = urls.get(i).getAttributes().getNamedItem("href").getNodeValue();
+
+            // Skip repeating vacancies and parse locations for them
+            Set<LocationDTO> locationDTOs = new HashSet<>();
+            while (jobId.equals(urls.get(i).getAttributes().getNamedItem("href").getNodeValue())) {
+                locationDTOs.add(createLocationDto(cities.get(i)));
+                i++;
+            }
+
+            // Variable i stopped in incorrect position
+            i--;
+            vacancies.add(createVacancy(createVacancyDTO(jobId, locationDTOs)));
+        }
+
+        return vacancies;
+    }
+
+    @SneakyThrows
+    private VacancyDTO createVacancyDTO(String jobUrl, Set<LocationDTO> locationDTOS) {
+        VacancyDTO vacancyDTO = new VacancyDTO();
+
+        HtmlPage page = downloadPage(STRIPE_URL + jobUrl);
+        DomElement title = (DomElement) page.getByXPath(VACANCY_TITLE_XPATH).get(0);
+        DomElement jobType = ((DomElement) page.getByXPath(VACANCY_JOB_TYPE_XPATH).get(0)).getNextElementSibling();
+        DomElement team = ((DomElement) page.getByXPath(VACANCY_TEAM_XPATH).get(0)).getNextElementSibling();
+
+        vacancyDTO.setDescription(createDescription(page));
+        vacancyDTO.setUrl(STRIPE_URL + jobUrl);
+        vacancyDTO.setJobId(jobUrl.replaceAll("\\D+", ""));
+        vacancyDTO.setTeam(team.getTextContent());
+        vacancyDTO.setLocation(locationDTOS);
+        vacancyDTO.setJobType(jobType.getTextContent());
+        vacancyDTO.setTitle(title.getTextContent());
+
+        return vacancyDTO;
+    }
+
+    private Vacancy createVacancy(VacancyDTO vacancyDTO) {
+        Vacancy vacancy = new Vacancy();
+
+        vacancy.setTitle(vacancyDTO.getTitle());
+        vacancy.setDescription(createFullDescription(vacancyDTO));
+        vacancy.setUrl(vacancyDTO.getUrl());
+        vacancy.setJobId(vacancyDTO.getJobId());
+        vacancy.setCompany(company);
+        processLocations(vacancy, vacancyDTO.getLocation());
+
+        return vacancy;
+    }
+
+    private void processLocations(Vacancy vacancy, Set<LocationDTO> locations) {
+        locations.forEach(location -> vacancy.addLocation(locationProcessor.processLocation(
+                    company, location.getCity(), location.getCountryCode()
+            ))
+        );
+    }
+
+    private String createDescription(HtmlPage page) {
+        List<DomNode> description = page.getByXPath(VACANCY_DESCRIPTION_BLOCK_XPATH);
+
+        StringBuilder stringBuilder = new StringBuilder();
+        description.forEach(
```

Comment:
I have a feeling that there's such method defined in the base Parser class. Probably you might reuse it instead of duplicating the code

Reply:
I didn't find this method in the Parser class

Reply:
You are right, I didn't commit this. Nevermind